docs/python/how-to-guides/use-uris.md

```diff
@@ -119,16 +202,45 @@ If we have the hostname of our colleague's machine, that can be used in place of
 
 If the machine on which a table exists is public, then consuming that table is done the same way as if it were a private network. All that's needed is the hostname/IP and table name.
 
-<!-- TODO:
+## Performance considerations
+
+When using URIs to share tables across instances, particularly over networks, there are several performance factors to consider:
+
+### Network impact
+
+- **Latency**: Table access over a network introduces latency that varies based on network conditions. For operations requiring low latency, consider co-locating instances when possible.
+- **Bandwidth**: Large tables or tables with frequent updates require more bandwidth. Network throughput can become a bottleneck for data-intensive operations.
+- **Connection reliability**: Unstable network connections can affect the reliability of table access. Implement appropriate error handling for network disruptions.
+
+### Table characteristics
+
+- **Table size**: Larger tables require more resources to transfer and update. The overhead increases with row count and column width.
+- **Update frequency**: Tables with high update frequencies consume more network resources. Each update must be synchronized across instances.
+- **Column types**: Tables with complex data types like large strings or nested structures may have higher overhead when shared across instances.
+
+### Optimization strategies
+
+- **Filter at source**: Apply filters on the source table before resolving it in the consumer instance to reduce data transfer volume.
+- **Project necessary columns**: Only include columns that are actually needed by the consumer to minimize network payload size.
+- **Pre-aggregate when possible**: For large tables, use aggregations at the source to reduce data volume while preserving analytical value.
+- **Minimize resolution frequency**: For latency-sensitive applications, cache resolved table references rather than repeatedly resolving the same URI.
+- **Use table snapshots**: When point-in-time consistency is required, use table snapshots instead of live updating tables.
```

Comment:
What is `point-in-time consistency`? How do snapshots help you achieve that over ticking tables?